flo-freezer/src/main/java/com/spotify/flo/freezer/PersistingContext.java

```diff
@@ -95,39 +97,48 @@ public PersistingContext(Path basePath, EvalContext delegate) {
     return promise.value();
   }
 
-  public static void serialize(Object object, Path file) throws Exception{
-    final Kryo kryo = new Kryo();
-    PackageRegistrar.all().apply(kryo);
-    kryo.register(java.lang.invoke.SerializedLambda.class);
-    kryo.register(ClosureSerializer.Closure.class, new ClosureSerializer());
-    kryo.addDefaultSerializer(java.lang.Throwable.class, new JavaSerializer());
-    kryo.getFieldSerializerConfig().setIgnoreSyntheticFields(false);
-
-    if (Files.exists(file)) {
-      throw new RuntimeException("File " + file + " already exists");
-    }
-
+  public static void serialize(Object object, Path file) throws Exception {
     try {
-      try (Output output = new Output(newOutputStream(file, WRITE, CREATE_NEW))) {
-        kryo.writeClassAndObject(output, object);
-      }
+      serialize(object, Files.newOutputStream(file, WRITE, CREATE_NEW));
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  public static void serialize(Object object, OutputStream outputStream) {
```

Comment:
Any reason we expose another two APIs?

Reply:
Yes. Then we can have unit tests for tasks, trying serialize and deserialize only using Streams and without using the file system.

These unit tests and/or calling `FloRunner.runTask` is important otherwise unserializeable flo tasks would only be noted at runtime in production.

Reply:
👍